ui/pages/asset/components/native-asset.tsx

```diff
@@ -29,7 +31,13 @@ const NativeAsset = ({ token, chainId }: { token: Token; chainId: Hex }) => {
   const { address } = useSelector(getSelectedInternalAccount);
   const rpcPrefs = useSelector(getRpcPrefsForCurrentProvider);
 
-  const selectedAccount = useSelector(getSelectedAccount);
+  const caipChainId = isCaipChainId(chainId)
+    ? chainId
+    : formatChainIdToCaip(chainId);
+  // TODO BIP44: The new selector returns the accountId, when BIP44 is fully enabled we can fetch the asset higher up and ensure it's passed here
```

Comment:
I have tried refactoring by moving some logic to asset.tsx, which is the parent of both this and token-asset.tsx.

Even correcting the type of `chainId` from `Hex` to `Hex | CaipChainId` causes a large amount of type error, so a major refactor is needed.

There are also no tests for neither `asset.tsx`, `token-asset.tsx` or `native-asset.tsx`.

This could be so much simpler with a top component that acts as guard that checks that there is a chain and an account (which could be extracted from the new selector), but that's too much of a refactor for a fix.

Added ticket for cleaning up after BIP44: https://consensyssoftware.atlassian.net/browse/ASSETS-1345